packages/bsky/src/notifications.ts

```diff
@@ -0,0 +1,285 @@
+import axios from 'axios'
+import Database from './db'
+import { Notification } from './db/tables/notification'
+import { AppBskyEmbedImages, AtUri } from '@atproto/api'
+import { Insertable } from 'kysely'
+
+const PUSH_NOTIF_SERVER_BASE_URL = 'https://push.bsky.app'
+const PUSH_NOTIF_SERVER_ENDPOINT = '/api/push'
+export const PUSH_NOTIF_SERVER_URL =
+  PUSH_NOTIF_SERVER_BASE_URL + PUSH_NOTIF_SERVER_ENDPOINT
+export const BSKY_APP_ID = 'xyz.blueskyweb.app'
+type Platform = 'ios' | 'android' | 'web'
+type PushNotification = {
+  tokens: string[]
+  platform: 1 | 2 // 1 = ios, 2 = android
+  title: string
+  message: string
+  topic: string
+  data?: {
+    [key: string]: string
+  }
+}
+type InsertableNotif = Insertable<Notification>
+
+export class NotificationServer {
+  constructor(public db: Database) {}
+  static creator() {
+    return (db: Database) => new NotificationServer(db)
+  }
+
+  async getUserTokens(did: string) {
+    const userTokens = await this.db.db
+      .selectFrom('notification_push_token')
+      .where('did', '=', did)
+      .selectAll()
+      .execute()
+
+    return userTokens
+  }
+
+  async prepareNotifsToSend(notifications: InsertableNotif[]) {
+    const notifsToSend: PushNotification[] = []
+
+    for (const notif of notifications) {
+      const { did: userDid } = notif
+      const userTokens = await this.getUserTokens(userDid)
+      // if user has no tokens, skip
+      if (!userTokens || userTokens.length === 0) {
+        continue
+      }
+
+      for (const t of userTokens) {
+        const { appId, platform, token } = t
+        // get title and message of notification
+        const attr = await this.getNotificationDisplayAttributes(notif)
+        // if no title or body, skip
+        if (!attr) {
+          continue
+        }
+        const { title, body } = attr
+        if (platform === 'ios') {
+          notifsToSend.push({
+            tokens: [token],
+            platform: 1,
+            title: title,
+            message: body,
+            topic: appId,
+            data: {
+              reason: notif.reason,
+              recordUri: notif.recordUri,
+              recordCid: notif.recordCid,
+            },
+          })
+        } else if (platform === 'android') {
```

Comment:
Nit: could use a ternary for the value of `platform` prop here and dry it up a bit

Reply:
We will need to create a new interface for web notifs so that is why I made it more verbose.

Reply:
sry real quick to circle back on this - how different is the web interface? 

i'm also inclined to simplify this function just a bit

honestly even something like
```ts
if(platform === 'ios' || platform === 'android'){
  // do the format w a ternary
} else {
  // handle web
}
```

Reply:
The web interface will be completely different. Happy to clean this up a bit like you suggested but I was optimizing for readability.